// hcs.js/src/selfCheckList.js
const request = require('./request')

module.exports = async (endpoint, token) => {
    const classData = await request('/joinClassList', 'POST', {}, endpoint, {'Authorization': token});
    
    if(classData['classList'] === undefined) return {
        success: false,
        message: '해당 선생님이 관리하는 반 정보가 없습니다.'
    }

    if(classData['classList'].length < 1) return {
        success: false,
        message: '해당 선생님이 관리하는 반 정보가 없습니다.'
    }
    let result = {'classInfo':{}, 'userInfo':{}};
    result['classInfo'] = { 'grade':classData['classList'][0]['grade'], 'class':classData['classList'][0]['classCode'] };
    const list = await request('/join', 'POST', classData.classList[0], endpoint, {'Authorization': token});
    /*
    {
        orgCode: '',
        inveYmd: '',
        grade: '', //학년
        classCode: '', //반
        serveyTime: '',
        name: '', //이름
        userPNo: '',
        surveyYn: 'Y', //참여여부
        rspns01: '',
        rspns02: '',
        rspns09: '',
        rspns00: '',
        deviceUuidYn: 'Y', //공식앱 설치여부
        registerDtm: '', //자가진단제출 시간
        stdntCnEncpt: '', //번호
        upperUserName: '' //이름
        }
    */
    for(let i=0; i<list.joinList.length; i++) {
        result['userInfo'][list.joinList[i]['stdntCnEncpt']] = list.joinList[i]['surveyYn'];
    }
    return {
        success: true,
        data: result
    }
}
